external/mmdetection/detection_tasks/apis/detection/openvino_task.py

```diff
@@ -576,18 +577,43 @@ def infer(
                     representation_vector, model=self.model
                 )
 
+            add_saliency_map = True
             if add_saliency_map and saliency_map is not None:
-                saliency_map = get_actmap(
-                    saliency_map, (dataset_item.width, dataset_item.height)
-                )
-                saliency_map_media = ResultMediaEntity(
-                    name="Saliency Map",
-                    type="saliency_map",
-                    annotation_scene=dataset_item.annotation_scene,
-                    numpy=saliency_map,
-                    roi=dataset_item.roi,
-                )
-                dataset_item.append_metadata_item(saliency_map_media, model=self.model)
+                if saliency_map.ndim == 2:
+                    # Single saliency map per image, support e.g. EigenCAM use case
+                    actmap = get_actmap(
+                        saliency_map, (dataset_item.width, dataset_item.height)
+                    )
+                    saliency_media = ResultMediaEntity(
+                        name="Saliency Map",
+                        type="saliency_map",
+                        annotation_scene=dataset_item.annotation_scene,
+                        numpy=actmap,
+                        roi=dataset_item.roi
+                    )
+                    dataset_item.append_metadata_item(saliency_media, model=self.model)
+                elif saliency_map.ndim == 3:
+                    # Multiple saliency maps per image (class-wise saliency map)
+                    labels = self.task_environment.get_labels(include_empty=True)
+                    num_saliency_maps = saliency_map.shape[0]
+                    if num_saliency_maps == len(labels) + 1:
+                        # Include the background as the last label
+                        labels.append(LabelEntity('background', Domain.DETECTION))
+                    for class_id, class_wise_saliency_map in enumerate(saliency_map):
+                        actmap = get_actmap(
+                            class_wise_saliency_map, (dataset_item.width, dataset_item.height)
+                        )
+                        class_name_str = labels[class_id].name
+                        saliency_media = ResultMediaEntity(
+                            name=class_name_str,
```

Comment:
Why don't you align with classification?
![image](https://user-images.githubusercontent.com/4629606/204970738-794387f1-bad7-47ae-9c90-089f225877a7.png)

Reply:
Thanks for notice. 
Geti UI already have static `Saliency map`, so it is clear for the user that it is a saliency map.
Thus, it was decided to replace `f"Saliency Map: {name=class_name_str}"` with `name=class_name_str`.
So that in the drop down menu we have just the names of the the classes.

I will align classification with detection.